debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -378,18 +399,54 @@ protected void handleServerHeartbeat(Event event) {
     /**
      * Handle the supplied event that signals that an out of the ordinary event that occurred on the master. It notifies the slave
      * that something happened on the master that might cause data to be in an inconsistent state.
-     * 
+     *
      * @param event the server stopped event to be processed; may not be null
      */
     protected void handleServerIncident(Event event) {
-        logger.trace("Server incident: {}", event);
+        if (event.getData() instanceof EventDataDeserializationExceptionData) {
+            EventDataDeserializationExceptionData data = event.getData();
+
+            // TODO make it an option
+            EventDeserializationFailureHandlingMode deserializationFailureHandling = EventDeserializationFailureHandlingMode.FAIL;
+            EventHeaderV4 eventHeader = (EventHeaderV4) data.getCause().getEventHeader(); // safe cast, instantiated that ourselves
+
+            if(deserializationFailureHandling == EventDeserializationFailureHandlingMode.FAIL) {
+                // logging some additional context but not the exception itself, this will happen in handleEvent()
+
+                logger.error(
+                        "Error while deserializing binlog event at offset {}." + System.lineSeparator() +
+                        "Use the mysqlbinlog tool to view the problematic event: `mysqlbinlog --start-position={} --stop-position={} --verbose {}",
+                        source.offset(),
+                        eventHeader.getPosition(),
+                        eventHeader.getNextPosition(),
+                        source.binlogFilename()
+                );
+
+                throw new RuntimeException(data.getCause());
+            }
+            else if(deserializationFailureHandling == EventDeserializationFailureHandlingMode.WARN) {
+                logger.warn(
+                        "Error while deserializing binlog event at offset {}." + System.lineSeparator() +
+                        "This exception will be ignored and the event be skipped." + System.lineSeparator() +
+                        "Use the mysqlbinlog tool to view the problematic event: `mysqlbinlog --start-position={} --stop-position={} --verbose {}",
+                        source.offset(),
+                        eventHeader.getPosition(),
+                        eventHeader.getNextPosition(),
```

Comment:
Won't this show the *next* binlog event that is after the problematic one?

Reply:
Hum, no, I don't think so. It uses getPosition() as the start and and nextPosition() as the stop position. Copying the resulting output and invoking _mysqlbinlog_ pointed me to the right offset.